client/src/components/Breadcrumbs.jsx

```diff
@@ -1,25 +1,28 @@
 import './Breadcrumbs.css'
 
+const pathToBreadcrumbs = (currentPath) => {
+  const paths = currentPath.split('/').slice(1)
+  const breadcrumbs = [{ name: 'home', path: '/' }]
+  let currentFullPath = ''
+
+  paths.forEach((name) => {
+    currentFullPath += `/${name}`
+    breadcrumbs.push({ name, path: currentFullPath })
+  })
+
+  return breadcrumbs
+}
+
 const Breadcrumbs = ({ currentPath, fetchFiles }) => {
+  const breadcrumbs = pathToBreadcrumbs(currentPath)
   return (
     <nav aria-label='Breadcrumb' className='breadcrumbs'>
       <ul className='breadcrumb'>
-        <li className='breadcrumb-item' onClick={() => fetchFiles('/')}>
-          home
-        </li>
-        {currentPath !== '/' &&
-          currentPath
-            .split('/')
-            .slice(1)
-            .map((name, index, array) => {
-              const path = `/${array.slice(0, index + 1).join('/')}`
-
-              return (
-                <li key={index} className='breadcrumb-item' onClick={() => fetchFiles(path)}>
-                  <span>{name}</span>
-                </li>
-              )
-            })}
+        {breadcrumbs.map((breadcrumb, index) => (

```

Comment:
destruct  "breadcrumb" into {name, path}

Reply:
I did it)